DOCS: Document new --differential option to gprecoverseg (#15230)
greenplum-db/gpdb: gpdb-doc/markdown/admin_guide/highavail/topics/g-understanding-segment-recovery.html.md

> **Note** After a failed incremental recovery attempt you must perform a full recovery.

Whenever possible, you should perform an incremental recovery rather than a full recovery, as incremental recovery is substantially faster. 

> If you **do** need to perform a full recovery, you can speed up its progress with `gprecoverseg`'s `--differential` option, which causes `gprecoverseg` to skip recovery of any files and directories that are unchanged since the last time `gprecoverseg` ran.

Reviewer: ```suggestion
If you **do** need to perform **inplace** full recovery, you can speed up its progress with `gprecoverseg`'s `--differential` option, which causes `gprecoverseg` to skip recovery of any files and directories that are unchanged.
```

As differential method  should be only used for inplace recoveries. Should not be used if doing full recovery of a segment to new host

Reviewer: Line 88 should be changed to "Greenplum database can perform three types of recovery: incremental, differential and full. The default is incremental."

Author: Also, this doesn't make sense to me: "As differential method should be only used for inplace recoveries. Should not be used if doing full recovery of a segment to new host"

the --differential flag is precisely for speeding up full recovery. So what do you mean by "differential method"?

Author: @Annu149 Are you using "differential recovery" to mean a full recovery that invokes the --differential flag? 

We need to be careful not to muddy the terminology here.

Reviewer: Full recovery can be done two ways: inplace and new host/port/data_dir. So when user wants to do inplace full 
recovery then only differential can be used

Reviewer: Also, can you please add a note that 

- differential recovery is not supported when using input configuration file (gprecoverseg -i) 
- --differential flag should not be used with -i, -o and -p flags

Author: @Annu149 
There's a larger issue here which is that beyond the world of Greenplum database folks talk about 3 types of recovery: full, incremental, and differential. 

So part of fulfilling this JIRA story / this new feature is to update the narrative docs to lay out these 3 possibilities. Because right now, the segment recovery documentation only lays out 2 possibilities: incremental and full.

I'd rather not cast --differential as an aspect of full, which it sounds like you're doing. Rather, it is a 3rd, distinct type of recovery. Of course, feel free to disagree.
